auth/data/src/main/java/com/nolawiworkineh/auth/data/NamePatternValidator.kt

```diff
@@ -0,0 +1,22 @@
+package com.nolawiworkineh.auth.data
+
+import com.nolawiworkineh.auth.domain.PatternValidator
+
+object NamePatternValidator : PatternValidator {
+    private const val MIN = 4
+    private const val MAX = 50
+    override fun matches(value: String): Boolean {
+        val trimmed = value.trim()
+        val words = trimmed.split("")
+
+        val hasValidLength = trimmed.length in MIN..MAX
+        val hasTwoWords = words.size >= 2
```

Comment:
The requirements document does not specify that a name has to have 2 words 👀 
When working with real projects, it is crucial to make sure you understand the requirements carefully so that you implement exactly what the client wants. Most clients would return to you saying there is a bug with the name validation, because they expect just 4 characters should pass the validation.